### PR TITLE
Various fixes and cleanups

### PR DESCRIPTION
**Pull request by xsawyerx**
[This falls under my Pull Request Challenge, extending my PR for Dancer2-Plugin-DBIC into this distribution.]

This includes similar fixes as those submitted to Dancer2-Plugin-DBIC. You can view this as "Step 2 out of 4".

The fixes and improvements, in summary, include:
* Provide META.json file in the dist.
* Provide provides entry in the metadata.
* Code cleanups and simplification.
* Explicitly use `DBIx::Class::Schema::Loader` and declare it appropriately, resolving #18 in Dancer2-Plugin-DBIC.
* Replace `die` calls with `croak`.

### CODE REVIEW

File: lib/DBICx/Sugar.pm [hunk 1]
```diff
@@ -95,23 +96,21 @@ sub schema {
             $schema = $schema_class->connect( @conn_info );
         }
     } else {
-        my $dbic_loader = 'DBIx::Class::Schema::Loader';
-        eval { load $dbic_loader };
-        die "You must provide a schema_class option or install $dbic_loader."
-            if $@;
-        $dbic_loader->naming( $options->{schema_loader_naming} || 'v7' );
+        eval { require DBIx::Class::Schema::Loader; 1; }
```
Comment (ironcamel) [inline lib/DBICx/Sugar.pm]: Why did you change this from `eval { load $dbic_loader };`?

Comment (xsawyerx) [reply lib/DBICx/Sugar.pm]: The `eval` is a bit clearer and I always prefer using barewords for `require` where possible.

I changed it because I hoped it will be clearer for others to read. This was missed in a comment on Dancer2-Plugin-DBIC in which recognizing this was used was missed.

Feel free to skip this if you don't like it.

**Comment by hvoers** [comment]
On Sat, 11 Feb 2017, Sawyer X wrote:

>> @@ -95,23 +96,21 @@ sub schema {
>             $schema = $schema_class->connect( @conn_info );
>         }
>     } else {
> -        my $dbic_loader = 'DBIx::Class::Schema::Loader';
> -        eval { load $dbic_loader };
> -        die "You must provide a schema_class option or install $dbic_loader."
> -            if $@;
> -        $dbic_loader->naming( $options->{schema_loader_naming} || 'v7' );
> +        eval { require DBIx::Class::Schema::Loader; 1; }
>
> The `eval` is a bit clearer and I always prefer using barewords for `require` where possible.

You also change "load" to "require".

> I changed it because I hoped it will be clearer for others to read. This was missed in a comment on Dancer2-Plugin-DBIC in which recognizing this was used was missed.

If you refer to my comment, I dit not mis it.

Niether Dancer2-Plugin-DBIC nor DBICx-Sugar has a
"use DBIx::Class::Schema::Loader;" statement.

So, DBIx::Class::Schema::Loader is not a requirement,
just an option. Prevent the dependency and just die.

**Comment by xsawyerx** [comment]
I know I changed the load to require, I have explained it here: https://github.com/ironcamel/DBICx-Sugar/pull/4/commits/4f80a6f9ef3c994ffda9991d7fe7839307b6abbf

I have made it reported optional appropriately here: https://github.com/ironcamel/DBICx-Sugar/pull/4/commits/9e527d272a25e4be0a64454d8b258211f171211e

Anywho, feel free to skip this one, like I said. I personally think it's cleaner to use `require` directly when possible, but that's just my personal preference. :)

**Comment by hvoers** [comment]
Aha. Okay.

File: lib/DBICx/Sugar.pm (hunk 1 above)
Comment (ironcamel) [reply lib/DBICx/Sugar.pm]: I think the old code is cleaner. Please change this part back. Also, please make sure all lines are <= 80 characters long.

**Comment by xsawyerx** [comment]
I reverted that specific commit. I had made no other changes that pass 80 characters.
